### PR TITLE
Drop support for Python 2.7, 3.5, 3.6

### PR DESCRIPTION
**Pull request by dataflake**
Python 3.11 still fails with a cryptic error coming out of a 🤮 doctest 🤮. I am not sure how to diagnose the issue.

### CODE REVIEW

File: src/five/formlib/tests/forms.txt [hunk 1]
```diff
@@ -199,20 +195,6 @@ the content, this means that no IObjectModified event should have been fired
...
-for Python 3. The following function converts our input values
-accordingly:
-
-  >>> def to_input(v):
-  ...     if six.PY2:
-  ...         return v.encode("utf-8")
-  ...     else:
-  ...         return v
-
 Even though ZPublisher does not support unicode, automatically
```
Comment (d-maurer) [inline src/five/formlib/tests/forms.txt:198]: This is no longer true: with Python 3, `ZPublisher` works with `str` (which is unicode), decoded under the assumption that `default_encoding` is the likely encoding (which can be changed with an encoding modifier).

File: src/five/formlib/tests/test_formlib.py [hunk 2]
```diff
@@ -68,8 +68,7 @@ def http_request(url, form_parts=None, body=None, auth=None):
     """perform HTTP request from given parameters.
 
     The primary purpose of this auxiliary function is to compute
-    the `Content-Length` header. While Python 2's `cgi` module
-    seems resilient to a wrong `Content-Length`, Python 3's `cgi` fails
+    the `Content-Length` header. Python 3's `cgi` fails
```
Comment (d-maurer) [inline src/five/formlib/tests/test_formlib.py:71]: `Zope` no longer uses `cgi`. However, if a `Content-Length` header is present, it must still be correct.

**Review by d-maurer — COMMENTED** [review]
Ok. But the tailing Python 3.11 test needs to be addressed (apparently, there is a change regarding `TemporaryFile`).

**Comment by icemac** [comment]
@dataflake The Python 3.11 error is https://github.com/python/cpython/issues/90113. See https://github.com/zopefoundation/zope.testbrowser/pull/99/commits/1bf73619edd018f4c8da9915c2e5f1cfac675a7c#diff-30acd2539f6fc012a8e0dcf153e8be0f7f514e065f68b2d5291cf518cc45769b as an example how I fixed this in `zope.testbrowser`.

**Review by d-maurer — APPROVED** [review]

**Pull request merged**